### DISCUSSION TITLE
Wrapping only strings with whitespace in quotes when converting to a properties file

### DISCUSSION THREAD

**Discussion by embee1981** [mikefarah/yq#1138] [Q&A] [answered]
Hi all

I'm trying achieve wrapping of string values with quotes if they have whitespace when converting to a props file. 

I'm using the following:

`yq -o=props '(.[] | select(tag == "!!str" and test("\s"))) |= ("\"" + . + "\"")' myfile.yaml`

which works with yaml files like:
```
bob: hello alice
hello: world
```
producing
```
bob = "hello alice"
hello = world
```

I'm doing something wrong however as when I run with something like the following

```
bob: hello alice
hello: false
blah: 123
goodbye:
  cruel: world
  more:
    nesting: here
```

It doesn't work and I get one of the following errors, depending on what I leave in the yaml file:
`Error: cannot match with !!bool, can only match strings. Hint: Most often you'll want to use '|=' over '=' for this operation`
`Error: cannot match with !!int, can only match strings. Hint: Most often you'll want to use '|=' over '=' for this operation`
`Error: cannot match with !!map, can only match strings. Hint: Most often you'll want to use '|=' over '=' for this operation`

What's the best way to achieve this?

UPDATE
The following appears to to what I want, assuming there are no tab chars present ;-) :
`yq -o=props '(.. | select(tag == "!!str" and . =="* *")) |= ("\"" + . + "\"")' myfile.yaml`
I think only wildcards can be used here though and not other regex'y things like \s ?

**Comment by mikefarah** [comment, marked as answer]
This is a bug in `yq` - it's evaluating the `test` operator in the `and` expression even though the first argument is false (i.e. it's a boolean). 

I'll work on a fix, till then you can use the workaround you have, or have multiple selects (and avoid `and` :/)

**Reply by mikefarah** [reply]
Fixed in 4.23.1
